tests/components/ness_alarm/test_init.py

```diff
@@ -206,6 +211,17 @@ class MockArmingState(Enum):
     TRIGGERED = "TRIGGERED"
 
 
+class MockArmingMode(Enum):
+    """Mock nessclient.ArmingMode enum."""
+
+    ARMED_AWAY = "ARMED_AWAY"
+    ARMED_HOME = "ARMED_HOME"
+    ARMED_DAY = "ARMED_DAY"
+    ARMED_NIGHT = "ARMED_NIGHT"
+    ARMED_VACATION = "ARMED_VACATION"
+    ARMED_HIGHEST = "ARMED_HIGHEST"
+
+
```

Comment:
Why do you mock out the enum, that seems really error prone if the library changes?

Reply:
I'd suggest to remove mocking the enums in the tests:
```diff
diff --git a/tests/components/ness_alarm/test_init.py b/tests/components/ness_alarm/test_init.py
index 1f325d84af3..2c3ce177f54 100644
--- a/tests/components/ness_alarm/test_init.py
+++ b/tests/components/ness_alarm/test_init.py
@@ -2,6 +2,7 @@
 from enum import Enum
 from unittest.mock import MagicMock, patch

+from nessclient import ArmingMode, ArmingState
 import pytest

 from homeassistant.components import alarm_control_panel
@@ -86,7 +87,7 @@ async def test_dispatch_state_change(hass: HomeAssistant, mock_nessclient) -> No
     await hass.async_block_till_done()

     on_state_change = mock_nessclient.on_state_change.call_args[0][0]
-    on_state_change(MockArmingState.ARMING, None)
+    on_state_change(ArmingState.ARMING, None)

     await hass.async_block_till_done()
     assert hass.states.is_state("alarm_control_panel.alarm_panel", STATE_ALARM_ARMING)
@@ -176,16 +177,16 @@ async def test_dispatch_zone_change(hass: HomeAssistant, mock_nessclient) -> Non
 async def test_arming_state_change(hass: HomeAssistant, mock_nessclient) -> None:
     """Test arming state change handing."""
     states = [
-        (MockArmingState.UNKNOWN, None, STATE_UNKNOWN),
-        (MockArmingState.DISARMED, None, STATE_ALARM_DISARMED),
-        (MockArmingState.ARMING, None, STATE_ALARM_ARMING),
-        (MockArmingState.EXIT_DELAY, None, STATE_ALARM_ARMING),
-        (MockArmingState.ARMED, None, STATE_ALARM_ARMED_AWAY),
-        (MockArmingState.ARMED, MockArmingMode.ARMED_AWAY, STATE_ALARM_ARMED_AWAY),
-        (MockArmingState.ARMED, MockArmingMode.ARMED_HOME, STATE_ALARM_ARMED_HOME),
-        (MockArmingState.ARMED, MockArmingMode.ARMED_NIGHT, STATE_ALARM_ARMED_NIGHT),
-        (MockArmingState.ENTRY_DELAY, None, STATE_ALARM_PENDING),
-        (MockArmingState.TRIGGERED, None, STATE_ALARM_TRIGGERED),
+        (ArmingState.UNKNOWN, None, STATE_UNKNOWN),
+        (ArmingState.DISARMED, None, STATE_ALARM_DISARMED),
+        (ArmingState.ARMING, None, STATE_ALARM_ARMING),
+        (ArmingState.EXIT_DELAY, None, STATE_ALARM_ARMING),
+        (ArmingState.ARMED, None, STATE_ALARM_ARMED_AWAY),
+        (ArmingState.ARMED, ArmingMode.ARMED_AWAY, STATE_ALARM_ARMED_AWAY),
+        (ArmingState.ARMED, ArmingMode.ARMED_HOME, STATE_ALARM_ARMED_HOME),
+        (ArmingState.ARMED, ArmingMode.ARMED_NIGHT, STATE_ALARM_ARMED_NIGHT),
+        (ArmingState.ENTRY_DELAY, None, STATE_ALARM_PENDING),
+        (ArmingState.TRIGGERED, None, STATE_ALARM_TRIGGERED),
     ]

     await async_setup_component(hass, DOMAIN, VALID_CONFIG)
@@ -199,29 +200,6 @@ async def test_arming_state_change(hass: HomeAssistant, mock_nessclient) -> None
         assert hass.states.is_state("alarm_control_panel.alarm_panel", expected_state)


-class MockArmingState(Enum):
-    """Mock nessclient.ArmingState enum."""
-
-    UNKNOWN = "UNKNOWN"
-    DISARMED = "DISARMED"
-    ARMING = "ARMING"
-    EXIT_DELAY = "EXIT_DELAY"
-    ARMED = "ARMED"
-    ENTRY_DELAY = "ENTRY_DELAY"
-    TRIGGERED = "TRIGGERED"
-
-
-class MockArmingMode(Enum):
-    """Mock nessclient.ArmingMode enum."""
-
-    ARMED_AWAY = "ARMED_AWAY"
-    ARMED_HOME = "ARMED_HOME"
-    ARMED_DAY = "ARMED_DAY"
-    ARMED_NIGHT = "ARMED_NIGHT"
-    ARMED_VACATION = "ARMED_VACATION"
-    ARMED_HIGHEST = "ARMED_HIGHEST"
-
-
 class MockClient:
     """Mock nessclient.Client stub."""

@@ -269,15 +247,5 @@ def mock_nessclient():

     with patch(
         "homeassistant.components.ness_alarm.Client", new=_mock_factory, create=True
-    ), patch(
-        "homeassistant.components.ness_alarm.ArmingState", new=MockArmingState
-    ), patch(
-        "homeassistant.components.ness_alarm.ArmingMode", new=MockArmingMode
-    ), patch(
-        "homeassistant.components.ness_alarm.alarm_control_panel.ArmingState",
-        new=MockArmingState,
-    ), patch(
-        "homeassistant.components.ness_alarm.alarm_control_panel.ArmingMode",
-        new=MockArmingMode,
     ):
         yield _mock_instance
```

Reply:
Sounds good. I think this is relic from the original PR where this was introduced (https://github.com/home-assistant/core/pull/94351). IIRC it was done to avoid the need to have a static dependency on `nessclient`  in order to run the tests.

Reply:
It's fine to have that dependency; integration dependencies are automatically added to `requirements_test_all.txt` for integrations with tests.